models/base/scratch/default/snowplow_media_player_base_events_this_run.sql

```diff
@@ -0,0 +1,270 @@
+{%- endif %}
+-- unpacking the web page context entity
+{{ snowplow_utils.get_sde_or_context(var('snowplow__atomic_schema', 'atomic'), var('snowplow__context_web_page'), lower_limit, upper_limit) }},
+
+prep as (
+  select
+    ev.event_id,
+    pv.id as page_view_id,
+    ev.domain_sessionid,
+    ev.domain_userid,
+    ev.page_referrer,
+    ev.page_url,
+    ev.geo_region_name,
+    ev.br_name,
+    ev.dvce_type,
+    ev.os_name,
+    ev.os_timezone,
+    ev.derived_tstamp as start_tstamp,
+    ev.collector_tstamp,
+
+    -- unpacking the media player event
+    mpe.label as media_label,
+    mpe.type as event_type,
+
+    -- unpacking the media player object
+    round(mp.duration) as duration,
+    mp.current_time as player_current_time,
+    coalesce(mp.playback_rate, 1) as playback_rate,
+    case when mpe.type = 'ended' then 100 else mp.percent_progress end percent_progress,
+    mp.muted as is_muted,
+    mp.is_live,
+    mp.loop,
+    mp.volume,
+
+    -- combined media properties
+    {{ media_id_col(youtube_player_id='yt.player_id', media_player_id='me.html_id') }},
+    {{ media_player_type_col(youtube_player_id='yt.player_id', media_player_id='me.html_id') }},
+    {{ source_url_col(youtube_url='yt.url', media_current_src='me.current_src')}},
+    {{ media_type_col(media_media_type='me.media_type')}},
+    {{ playback_quality_col(
+      youtube_quality='yt.playback_quality',
+      video_width='ve.video_width',
+      video_height='ve.video_height'
+    )}},
+
+    dense_rank() over (partition by domain_sessionid order by derived_tstamp) AS event_in_session_index
+
+    from events_this_run ev
+
+
+    -- youtube context entity
+    {% if var("snowplow__enable_youtube") %}
+        left join {{ var('snowplow__youtube_context') }} yt on ev.event_id = yt.youtube__id and ev.collector_tstamp = yt.youtube__tstamp
+    {%- endif %}
+    -- whatwg media context entity
+    {% if var("snowplow__enable_whatwg_media") %}
+    left join {{ var('snowplow__html5_media_element_context') }} me on ev.event_id = me.media_element__id and ev.collector_tstamp = me.media_element__tstamp
+    {%- endif %}
+    -- whatwg video context entity
+    {% if var("snowplow__enable_whatwg_video") %}
+    left join {{ var('snowplow__html5_video_element_context') }} ve on ev.event_id = ve.video_element__id and ev.collector_tstamp = ve.video_element__tstamp
+    {%- endif %}
+    -- media player event
+    left join {{ var('snowplow__media_player_event_context') }} mpe on ev.event_id = mpe.media_player_event__id and ev.collector_tstamp = mpe.media_player_event__tstamp
+    -- media player context entity
+    left join {{ var('snowplow__media_player_context') }} mp on ev.event_id = mp.media_player__id and ev.collector_tstamp = mp.media_player__tstamp
+    -- web page context entity
+    left join {{ var('snowplow__context_web_page') }} pv on ev.event_id = pv.web_page__id and ev.collector_tstamp = pv.web_page__tstamp
+
```

Comment:
Just to confirm, these are all single-object entities? So no chance for multiple objects per entity attached?